pkg/k8s/common.go

```diff
@@ -323,39 +324,82 @@ func getPortForwardUrl(config *rest.Config, namespace string, podName string) *u
 	}
 }
 
-func watchForReady(deploymentName *string, readyChan chan<- bool) {
+func watchForReady(deployment *appsv1.Deployment, readyChan chan<- bool) {
 	go func() {
+		lastMsg := ""
+
+		maxUnavailable := deployment.Spec.Strategy.RollingUpdate.MaxUnavailable.IntValue()
+		if maxUnavailable > 0 {
+			log.Warnf("RollingUpdate.MaxUnavailable: %v. This may prevent deployment failures from being detected. Set to 0 to ensure ProgressDeadlineInSeconds is enforced.", maxUnavailable)
+		}
+
+		progressDeadlineSeconds := int64(*deployment.Spec.ProgressDeadlineSeconds)
+		log.Infof("ProgressDeadlineInSeconds is currently %vs. It may take this long to detect a deployment failure.", progressDeadlineSeconds)
+		progressDeadlineSeconds += 5
+
+		watch, err := deploymentsClient.Watch(context.Background(), metav1.ListOptions{
+			LabelSelector:  labels.Set(deployment.Labels).String(),
+			TimeoutSeconds: &progressDeadlineSeconds,
+		})
+
+		if err != nil {
+			log.Error(err)
+			readyChan <- false
+			return
+		}
+		resultChan := watch.ResultChan()
+
 		for {
-			deployment, err := deploymentsClient.Get(context.Background(), *deploymentName, metav1.GetOptions{})
-			if err != nil {
-				log.Errorf("Failed fetching deployment; %v", err)
+			event, ok := <-resultChan
+			if !ok {
+				log.Error("Timeout exceeded waiting for deployment to be ready")
+				watch.Stop()
 				readyChan <- false
 				return
 			}
+
+			deployment, ok := event.Object.(*appsv1.Deployment)
+			if !ok {
+				log.Warn("Watch Received Not Deployment")
+				continue
+			}
+
 			msg, done, err := deploymentStatus(deployment)
 
 			if done {
-				print("\n")
+				watch.Stop()
 				log.Info(msg)
 				readyChan <- true
 				return
 			} else if err != nil {
-				print("\n")
+				watch.Stop()
 				log.Errorf("Failed deploying tunnel sidecar; %v", err)
 				readyChan <- false
 				return
 			} else {
-				print(".")
-				time.Sleep(time.Millisecond * 300)
+				if lastMsg != msg {
+					log.Info(msg)
```

Comment:
```suggestion
					log.Debug(msg)
```

Reply:
I can make this change but I'm not aware of any way to enable verbose logging for the logger used by the `k8s` package. Unless I'm mistaken, the current `-v` flag only affects the logger configured in the `cmd` package.